docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,111 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices Osds so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class.
+- UI will create the StorageAutoScale CR with a unique name i.e. <storageclusterName>-<deviceClassName>.
+  
+    ```
+    apiVersion: ocs.openshift.io/v1
+    kind: StorageAutoScaling
+    metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    spec:
+        storageCluster: 
+            name: <storageclusterName>
+        deviceClassName: ssd
+        deviceClassStorageCapacityLimit: <capacity>
+        osdScalingThresholdPercent: 70%  # should be less than osd nearfull percentage 
```

Comment:
Nit: With percent in name, this can be an int type IMO

```suggestion
        osdScalingThresholdPercent: 70  # should be less than osd nearfull percentage 
```